keystore/src/main/java/org/wildfly/security/keystore/KeyStoreUtil.java

```diff
@@ -134,13 +146,51 @@ private static KeyStore tryLoadKeystore(final Supplier<Provider[]> providers, fi
         return null;
     }
 
+    public static KeyStore loadPemAsKeyStore(FileInputStream is, char[] password) throws KeyStoreException, IOException {
+        KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        try {
+            keyStore.load(null);
+        } catch (Exception e) {
+            // won't happen
+        }
+        // try to load it as a PEM
+        PrivateKey pk = null;
+        List<Certificate> certificates = new ArrayList<>();
+        // Reading all of the file should not be an issue
+        byte[] pem = new byte[(int) is.getChannel().size()];
+        is.read(pem);
+        for (Iterator<PemEntry<?>> it = Pem.parsePemContent(CodePointIterator.ofUtf8Bytes(pem)); it.hasNext(); ) {
+            Object entry = it.next().getEntry();
+            if (entry instanceof PrivateKey) {
+                // Private key
+                pk = (PrivateKey) entry;
+            } else if (entry instanceof Certificate) {
+                // Certificate
+                Certificate certificate = (Certificate) entry;
+                certificates.add(certificate);
+            }
+        }
+        if (pk != null) {
+            // A keystore
+            keyStore.setKeyEntry("key", pk, password, certificates.toArray(new Certificate[0]));
```

Comment:
Instead of using "key" as the alias, could we use something like `getSubjectX500Principal().getName()` instead?

Reply:
A `PrivateKey` doesn't really have that, does it ?

Reply:
Or you mean I should use the name of the first cert in the chain ?

Reply:
Right, you could get this from the first cert in the chain.